rq/job.py

```diff
@@ -1248,6 +1254,11 @@ def delete(
         if delete_dependents:
             self.delete_dependents(pipeline=pipeline)
 
+        if self.batch_id:
+            from .batch import Batch
+
+            connection.delete(Batch.get_key(self.batch_id), self.id)  # Delete job from batch
```

Comment:
`connection.delete()` will delete the batch object from Redis. We should use `srem` instead. This command should also be pipelined.

Reply:
Also, mind adding a test for this? Deleting a job should not automatically delete batch if there are other jobs in the batch.

Reply:
To be safe, I think we should do `Batch().delete(job, pipeline=pipeline)`

Reply:
Done. I believe I do have a test for this in [`test_batch.py`](https://github.com/eswolinsky3241/rq/blob/1b40dd302fa0b39bd286f521d0e00b18dfb7bd78/tests/test_batch.py#L66). Let me know if it needs to be improved



























@selwin, I may be confused, connection is a pipeline if one is passed in correct? I think I'm testing this in [`test_batch.py`](https://github.com/eswolinsky3241/rq/blob/1b40dd302fa0b39bd286f521d0e00b18dfb7bd78/tests/test_batch.py#L66)